corehq/apps/reports/standard/deployments.py

```diff
@@ -59,113 +59,145 @@ class ApplicationStatusReport(DeploymentsReport):
             DataTablesColumn(_("Last Sync"),
-                             sort_type=DTSortType.NUMERIC),
+                             prop_name='reporting_metadata.last_syncs.sync_date',
+                             alt_prop_name='reporting_metadata.last_sync_for_user.sync_date'),
             DataTablesColumn(_("Application"),
-                             help_text=_("Displays application of last submitted form")),
+                             help_text=_("Displays application of last submitted form"),
+                             prop_name='reporting_metadata.last_builds.app_id',
+                             alt_prop_name='reporting_metadata.build_version_app'),
             DataTablesColumn(_("Application Version"),
                              help_text=_("""Displays application version of the last submitted form;
                                          The currently deployed version may be different."""),
-                             sort_type=DTSortType.NUMERIC),
+                             prop_name='reporting_metadata.last_builds.build_version',
+                             alt_prop_name='reporting_metadata.last_build.build_version'),
             DataTablesColumn(_("CommCare Version"),
                              help_text=_("""Displays CommCare version the user last submitted with;
                                          The currently deployed version may be different."""),
-                             sort_type=DTSortType.NUMERIC),
+                             prop_name='reporting_metadata.last_submissions.commcare_version',
+                             alt_prop_name='reporting_metadata.last_submission_for_user.commcare_version'),
         )
         headers.custom_sort = [[1, 'desc']]
         return headers
 
+    def get_sorting_block(self):
+        sort_prop_name = 'prop_name' if self.selected_app_id else 'alt_prop_name'
+        res = []
+        #the NUMBER of cols sorting
+        sort_cols = int(self.request.GET.get('iSortingCols', 0))
+        if sort_cols > 0:
+            for x in range(sort_cols):
+                col_key = 'iSortCol_%d' % x
+                sort_dir = self.request.GET['sSortDir_%d' % x]
+                col_id = int(self.request.GET[col_key])
+                col = self.headers.header[col_id]
+                sort_prop = getattr(col, sort_prop_name) or col.prop_name
+                sort_dict = {sort_prop: sort_dir}
+                res.append(sort_dict)
+        if len(res) == 0 and self.default_sort is not None:
+            res.append(self.default_sort)
+        return res
+
     @property
     @memoized
     def selected_app_id(self):
         return self.request_params.get(SelectApplicationFilter.slug, None)
 
-    @property
-    @memoized
-    def users(self):
-        mobile_user_and_group_slugs = self.request.GET.getlist(LocationRestrictedMobileWorkerFilter.slug)
+    @quickcache(['app_id'], timeout=10 * 60)
```

Comment:
can cache this for way longer i suspect. i don't think it matters if the name is wrong